src/main/java/com/google/sps/servlets/DatasetNamesServlet.java

```diff
@@ -41,24 +42,13 @@
 
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
-    Query query = new Query("MetaData").addSort("timestamp", SortDirection.ASCENDING);
-
-    /* Get current user logged in to webapp */
+    // Get current user logged in to webapp
     UserService userService = UserServiceFactory.getUserService();
     String userEmail = userService.getCurrentUser().getEmail();
 
-    DatastoreService datastore = DatastoreServiceFactory.getDatastoreService();
-    PreparedQuery results = datastore.prepare(query);
-
-    List<String> userDatasets = new ArrayList<String>();
-    for (Entity entity : results.asIterable()) {
-      String email = (String) entity.getProperty("user-email");
-      if (email.equals(userEmail)) {
-        // Get this the dataset-name for this entry and store it
-        String dataset = (String) entity.getProperty("dataset-name"); 
-        userDatasets.add(dataset);
-      }
-    }
+    // Get the logged in user's datasets
+    DatastoreMetadataStore datastoreStorage = new DatastoreMetadataStore();
```

Comment:
MetadataStore datastoreStorage = ...
So that you only care about how it was implemented when you create the object. You can even make this a private final variable of the class.

Reply:
Ack. I've changed all places in my PR where I made this mistake